merge

avshi6116 opened the pull request.

Commit:
Implement different algorithm for subnet dropdown generation

Commit:
Fix potentially problematic typo ($$ instead of $)

Commit:
Merge branch 'master' into optimize-subnet-dropdown

marcioelias committed:
Update Tools.php

Change to get Addresses and Subnets from a Location using API.

Pectojin committed:
Consistency update of extfield names

All the other import options lower case every extfield, so it seems weird that ip_addr import has a couple of uppercase letters.

Pectojin committed:
Fix ip_addr and dns_name mappings

The frontend states that the fields should be named "Hostname" and "IP Address" in excel, but then fails to match them during import because they're called "ip_addr" and "dns_name" internally (and in the DB). 

This change allows using either the database format or the one displayed in the frontend in excel sheets for import.

Pectojin committed:
Updated the ip address template

Changed field from DNS Hostname to Hostname for consistency.
Also found a mistake in the code where subnet and VRF was assigned to the same column causing the resulting template to have no subnet column.

aharenaz committed:
Issue 1440 fix: CSV Import fails when delimiter is different from coma.

Commit:
Ensure password filds is correct file/size

Old versions had the passwords stored as a VARCHAR(32).

This ensure that the password column in the users table is the same size
and type as the current schema, allowing upgrades from old versions.

phpipam committed:
Disabled caching and JS/CSS minimization

phpipam committed:
Mod_rewrite test

ciphergloom committed:
Updated German languag pack

Added some more menu points and corrected some misspellings

GaryAllan committed:
Merge branch 'optimize-subnet-dropdown' of https://github.com/alexat/phpipam into for-miha

GaryAllan committed:
Bugfix: subnet_dropdown_print_available patch fixes

Avoid integer overflow on 32bit platforms.
Ensure $subnets is set to empty list if subnet has no slaves.

GaryAllan committed:
Bugfix: subnet_dropdown_print_available patch fixes

Be consistent with prior behaviour and display all available
subnets for 8 levels.

Display the first 8 available subnets on remaining levels.

GaryAllan committed:
Cleanup: Style and redundant code cleanup for subnet_dropdown_print_available()

GaryAllan committed:
Bugfix: subnet_dropdown_print_available patch fixes v2

Be consistent with prior behaviour and display all available subnets for 8 levels.

GaryAllan committed:
Merge pull request #1562 from GaryAllan/alexat-fixes

Performance fix for subnet_dropdown_print_available()

phpipam committed:
Fixed in valid subnet Id #41

phpipam committed:
Merge pull request #1560 from il3ctro/il3ctro-de_DE.UTF-8

Updated German languag pack

phpipam committed:
Merge pull request #1460 from bencaron/expand-password-field

Ensure password filds is correct file/size

phpipam committed:
Merge pull request #1458 from aharenaz/fix/cvs_import_separator

Issue 1440 fix: CSV Import fails when delimiter is different from coma.

phpipam committed:
Merge pull request #1363 from marcioelias/patch-1

Improvement on API Tools/Locations Controller

phpipam committed:
Fixed Checking the availability of HTTP_X_FORWARDED_HOST PR #1142

phpipam committed:
Merge branch 'master' of https://github.com/phpipam/phpipam

phpipam committed:
Merge pull request #1457 from Pectojin/import_ipaddr_frontend_improvements

Import ipaddr frontend improvements

GaryAllan committed:
Refactor the range split/exclusion algorithm provided by @alexat
into freespacemap functions to allow re-use in other code areas.

GaryAllan committed:
Fix repoted Scrutinizer issues in commit.

GaryAllan committed:
Merge pull request #1564 from GaryAllan/class_subnet_refactor

Refactor the range split/exclusion algorithm provided by @alexat

GaryAllan committed:
Restore menu-tools to the default dashboard.

GaryAllan committed:
Ignore local changes to the git submodule directories by default

This avoids accidentally including local submodule changes in PRs using "git add -a" and de-clutters "git status".
Changes to submodules can still be explictity added and commited if required:

	git add app/login/captcha
	git add functions/PHPMailer
	git add functions/php-saml

GaryAllan committed:
Merge pull request #1570 from GaryAllan/ignore_submodules

Ignore local changes to the git submodule directories by default

GaryAllan committed:
IP Address Import generted template fixes. Issue #1328

GaryAllan committed:
Verify IPs reside inside subnet for IP imports. Issue #1567

GaryAllan committed:
Subnet->verify_overlapping() argument order matters for IPv6 addresses

phpipam committed:
Merge pull request #1580 from phpipam/mod_rewrite

Mod rewrite merge

phpipam committed:
Changed gitignore for logo

avkolbasenko committed:
Update .htaccess

Options (Allow/Deny/Require) depend of Apache version. Uploaded logo not working with Apache 2.4

GaryAllan committed:
Implement IP Address Import: Search for matching subnet in all VRFs

GaryAllan committed:
Implement IP Address Import: Search for matching subnet in all VRFs

Cascade searchallvrfs value from preview to import action.

GaryAllan committed:
Cleanup redundant code in import-recompute-logic.php

phpipam committed:
Fixed empty content type for nginx

phpipam committed:
API fixes for new rewrites #41

phpipam committed:
Merge branch 'master' of https://github.com/phpipam/phpipam

GaryAllan committed:
Fix Net:Ping deprecated class constructor warnings

GaryAllan committed:
Fix Net:Ping deprecated constructor warnings

Methods with the same name as their class will not be constructors in a future version of PHP; Net_Ping has a deprecated constructor
Methods with the same name as their class will not be constructors in a future version of PHP; Net_Ping_Result has a deprecated constructor

GaryAllan committed:
Fix for #1461, #1583

phpipam committed:
Fixed #1455

phpipam committed:
Merge pull request #1582 from avkolbasenko/patch-1

Update .htaccess

phpipam committed:
Rewritten rewrite module, /api/ in nginx location not needed anymore

phpipam committed:
Merge branch 'master' of https://github.com/phpipam/phpipam

GaryAllan committed:
Update CHANGELOG

GaryAllan committed:
Add app/admin/import-export/upload/ folder to .gitignore.

GaryAllan committed:
Fix for #1594

GaryAllan committed:
Fix for #1594, pt2

GaryAllan committed:
Fix for #1594, pt3

GaryAllan committed:
class.Rewrite.php, Preserve $_GET array content for uri_passthroughs. Issue #41

    if(in_array($this->uri_parts[0], $this->uri_passthroughs)) {
        $this->get_params = $_GET;
    } else {
        foreach ($this->uri_parts as $k=>$l) {
            ...

GaryAllan committed:
Update widgets to work with new class.Rewrite model. #41, #1593

GaryAllan committed:
NAT module cleanup, fix php warnings. #1605

GaryAllan committed:
Fix: Section names containing spaces are not included in exports. #1566

phpipam committed:
Update CHANGELOG

phpipam committed:
Added bootstrp-select submodule

thomseddon committed:
use x-real-ip in logs

GaryAllan committed:
Bugfix: dashboard top10_ widgets #1593

Exclude subnets with no user access from top_subnets list in widgets.
Ensure generated JS and HTML data tables are aligned.

GaryAllan committed:
Bugfix+Performance:  Tools->fetch_top_subnets

Ensure fetch_top_subnets('IPv6',10,true) returns data for IPv6 hosts.

Update SQL to make full use of indexes.

GaryAllan committed:
Refactor API to use split/exclusion algorithm from @alexat
Now possible to find all available /32's of 10.0.0.0/8 in < 60s.

    Subnets controller
      /api/my_app/subnets/{id}/first_subnet/{mask}/
      /api/my_app/subnets/{id}/all_subnets/{mask}/

    Prefixes controller
      /api/my_app/prefix/{customer_type}/{ip_version}/{mask}/

phpipam committed:
Merge pull request #1613 from thomseddon/feature/log-x-real-ip

use x-real-ip in logs

Commit:
Fixed invalid key for UI ping check

GaryAllan committed:
Misc "variable does not exist" fixes.

GaryAllan committed:
class.Tools "variable does not seem to be defined for all execution paths" fixes.
Importing IPs loose Mac Addresses #1607

phpipam committed:
Added assets to gitignore

phpipam committed:
Merge branch 'master' of https://github.com/phpipam/phpipam

phpipam committed:
Added split_ip_custom_fields directive to split custom fields for IP addresses editing to new table

phpipam committed:
Added required IP fields

phpipam committed:
Changed DB field dns_name to hostname

GaryAllan committed:
BugFix: IPv4/IPv6 2^32-1 network boundary fixes

Ensure IPv4/IPv6 type identification and behaviour is consistent
and correct at 2^32-1 subnet and ip_addr boundaries.

    Max IPv4 subnet/ip_addr = 255.255.255.255, 4294967295

GaryAllan committed:
BugFix: switch back user, 404 Not Found error #1640

GaryAllan committed:
BugFix: "Only variables should be passed by reference" php warnings.

phpipam committed:
Added append_qsa to Rewrite class that appends query parameters - ?cLimit=x&cSie=10

GaryAllan committed:
BugFix: Don't cache partial objects

GaryAllan committed:
Bugfix: Don't double cache fetch_subnet_slaves()

phpipam committed:
Fixed fied escapting for ciscuits in custom fields

phpipam committed:
Merge branch 'master' of https://github.com/phpipam/phpipam

phpipam committed:
Merged #1257 PR

phpipam committed:
fixed #1626 - Custom Field adds linebreak after editing

GaryAllan committed:
Bugfix: identify_address('255.255.255.255') should return 'IPv4'

GaryAllan committed:
Bugfix: Don't cache fetch_multiple_objects() requested with
partial result fields (i.e not “*”).

Future requests for these objects will be pulled from the cache and
may require the missing fields. This is very difficult to debug!

GaryAllan committed:
Bugfix: Don't double cache results in fetch_subnet_slaves()

$slave array can be large (10,000+) and is already saved to
the cache by Common_functions::fetch_multiple_objects()

GaryAllan commented:
Hi @avshi6116 

Can you add a description to this pull request describing what it is for/what it does? Also can you re-create the pull request against the 'master' development branch?

Thanks

GaryAllan committed:
Merge pull request #1652 from GaryAllan/bugfixes

Misc bug fixes. No new features or changes to behavior.

GaryAllan committed:
Merge pull request #1625 from GaryAllan/find_api

Re-factor API to use split/exclusion algorithm from @alexat 
Provides performance improvement. No new features or changes to existing behavior.

GaryAllan committed:
Merge pull request #1654 from GaryAllan/deprecated_class_constructors

Fix Net:Ping deprecated class constructor warnings
Bug fix only. No new functionality implemented.

GaryAllan committed:
AJAX mastersubnet dropdown menu

+ Refactor mastersubnetsdropdown code into dedicated Menu class.

+ Populating the input[name='subnet'] form field will dynamically
  update the mastersubnet dropdown <select> list.

  When section strict mode is enabled only overlapping subnets and
  folders are displayed in the menu.

  When section strict mode is disabled an additional overlapping
  subnets sections is displayed at the top of the list.

GaryAllan committed:
scrutinizer fixes

GaryAllan committed:
Fix reported scrutinizer issues

github138 committed:
fix slow mac address vendor decoding for subnet detail view

github138 committed:
fix import encoding if necessary

GaryAllan committed:
Code Cleanup and bugfixes

- BugFix and cleanup for $Subnets->get_max_hosts() on 32 bit platforms.
    php > var_dump( $Subnets->get_max_hosts(0, 'IPv4', false) );
    int(0)
    php > var_dump( $Subnets->get_max_hosts(1, 'IPv4', false) );
    int(-2147483648)

- Cleanup $bmask[][][] code to provide two new self documenting functions.
  Calculates the network and broadcast addresses of a subnet of size $mask
  e.g 10.10.10.10/8 => 10.0.0.0 & 10.255.255.255 (but in decimal format)

    $Subnets->decimal_network_address($decimalIP, $mask)
    $Subnets->decimal_broadcast_address($decimalIP, $mask)

- Cleanup and document/comment $Subnets freespacemap functions.

- Remove the unused subnet_dropdown_ipv6_decimal_add_one($decimalIpv6) function.
  Function can be replaced by gmp_strval(gmp_add($decimalIpv6, 1));

GaryAllan committed:
PDO Bugfixes

- BugFix for findObjects(). Use backtick $result_fields_arr in implode()

    // set fields
    if($result_fields!="*") {
        $result_fields_arr = array();
        foreach ($result_fields as $f) {
            $result_fields_arr[] = "`$f`";
        }
        // implode
        $result_fields = implode(",", $result_fields);
    }

GaryAllan committed:
fix scrutinizer warning

It is not recommend to use PHP's short opening tag <?,
better use <?php, or <?= in case of outputting.

Short opening tags are disabled in PHP’s default configuration.
In such a case, all content of this file is output verbatim to the
browser without being parsed, or executed.

GaryAllan committed:
Fix php docblock for &$rowCount

GaryAllan committed:
Add function Subnets->fetch_overlapping_subnets

      /*
        * Fetches all subnets averlapping with CIDR
        *
        * @access public
        * @param  string       $cidr
        * @param  string|null  $method
        * @param  string|null  $value
        * @param  string|array $result_fields
        * @return array|false
        */
       public function fetch_overlapping_subnets ($cidr, $method=null, $value=null, $result_fields = '*')

GaryAllan committed:
Performance and usability fixes

- Performance fix for Subnets->verify_vrf_overlapping()
  switch to fetch_overlapping_subnets() to minimise the search.

- Update Subnets->verify_vrf_overlapping() to display the section
  name of overlapping entries.

GaryAllan committed:
Performance and usability fixes

    - Performance fix for Subnets->verify_subnet_overlapping()
      switch to fetch_overlapping_subnets() to minimise the search.

    - Update Subnets->verify_subnet_overlapping() to display the section
      name of overlapping entries.

GaryAllan committed:
fix scrutinizer warning

GaryAllan committed:
Cleanup, Add PDO::escape_result_fields($result_fields)

Move duplicated code into escape_result_fields($result_fields) and
apply to all use cases of $result_fields.

GaryAllan committed:
Bugfix: identify_address('::1:0:0/128') should return 'IPv6'

GaryAllan committed:
Cleanup and improve performance of long2ip6()

GaryAllan committed:
Bugfix: $Subnets->verify_overlapping('2000::/3','3000::/128') should return 'true'
Bugfix: $Subnets->verify_overlapping('::1:0:0/128','::1:0:0/128') should return 'true'
Performance: Use gmp_math functions for $Subnets->verify_overlapping() [5x speedup]
Cleanup: Merge duplicate code in verify_overlapping() & is_subnet_inside_subnet()

GaryAllan committed:
Merge branch 'github138-import' into devel

GaryAllan committed:
Cleanup: Create convert_encoding_to_UTF8() function for reuse

GaryAllan committed:
Merge branch 'github138-macvendor' into devel

GaryAllan committed:
Cleanup get_mac_address_vendor_details()

GaryAllan committed:
Merge branch 'masterdropdown_ajax' into devel

GaryAllan committed:
Cleanup: switch mastersubnet-dropdown to fetch_overlapping_subnets()
Cleanup: class.Menus.php

GaryAllan committed:
Fix reported scrutinizer issues

GaryAllan committed:
Fix reported scrutinizer issues

GaryAllan committed:
Merge pull request #1660 from GaryAllan/devel

Performance + bug fixes

GaryAllan commented:
Hi @avshi6116

Can you please re-create the pull request against the 'master' development branch and include description/details?

Thanks
